src/Core/Events.xs

```diff
@@ -102,7 +102,8 @@ void
 events_set_event_filter(callback)
 	SV* callback
 	CODE:
-		eventfiltersv = callback;
+		SvREFCNT_dec(eventfiltersv);
+		eventfiltersv = SvREFCNT_inc(callback);
```

Comment:
One could use SvREFCNT_inc_simple() instead of a generic SvREFCNT_inc(). It's faster and applicable here. From perlapi POD:

        "SvREFCNT_inc_simple" can only be used with expressions without side
        effects. Since we don't have to store a temporary value, it's
        faster.

Reply:
Thanks!  I've pushed that change.

I knew XS pretty well 20 years ago, but I'm rusty now ...